PyDSS/node_voltage_metrics.py

```diff
@@ -0,0 +1,384 @@
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, List
+
+from pydantic import BaseModel, Field
+
+from PyDSS.utils.simulation_utils import CircularBufferHelper
+
+
+class VoltageMetricsBaseModel(BaseModel):
+    class Config:
+        title = "VoltageMetricsBaseModel"
+        anystr_strip_whitespace = True
+        validate_assignment = True
+        validate_all = True
+        extra = "forbid"
+        use_enum_values = False
+
+
+class VoltageMetric1(VoltageMetricsBaseModel):
+    time_points: List[datetime] = Field(
+        title="time_points",
+        description="time points that contain voltages between ANSI A and ANSI B thresholds",
+    )
+    duration: timedelta = Field(
+        title="duration",
+        description="amount of time where metric 1 existed (len(time_points) * resolution)",
+    )
+
+
+class VoltageMetric2(VoltageMetricsBaseModel):
+    duration: timedelta = Field(
+        title="duration",
+        description="amount of time where a node experienced ANSI A violations",
+    )
+    duration_percentage: float = Field(
+        title="duration_percentage",
+        description="percentage of overall time",
+    )
+
+
+class VoltageMetric3(VoltageMetricsBaseModel):
+    time_points: List[datetime] = Field(
+        title="time_points",
+        description="time points where moving average voltages violated ANSI A thresholds",
+    )
+    duration: timedelta = Field(
+        title="duration",
+        description="amount of time where metric 2 existed (len(time_points) * resolution)",
```

Comment:
description="amount of time where metric 3 existed (len(time_points) * resolution)"